test/gleam/erlang/file_test.gleam

```diff
@@ -0,0 +1,44 @@
+import gleam/string
+import gleam/erlang/file
+
+pub fn successful_read_test() {
+  assert Ok(<<"Hello, World!\n":utf8>>) =
+    file.read_file("test/fixtures/success.txt")
+}
+
+pub fn unsuccessful_read_test() {
+  assert Error(file.Enoent) = file.read_file("test/fixtures/does_not_exist.txt")
+}
+
+pub fn successful_write_test() {
+  let path = tmp_path("write_test.txt")
+
+  assert Ok(Nil) = file.write_file(path, <<"Hello, World!":utf8>>)
```

Comment:
Does this succeed if the file was already there?

Reply:
I have restructured the tests such that they check for nonexistence before writing, meaning that we won't have spurious test successes if `write` is broken.